lib/rspec-puppet/support.rb

```diff
@@ -254,6 +275,12 @@ def build_catalog_without_cache(nodename, facts_val, trusted_facts_val, hiera_co
     end
 
     def stub_facts!(facts)
+      if facts['operatingsystem'] && facts['operatingsystem'].to_s.downcase == 'windows'
+        Puppet::Util::Platform.pretend_to_be :windows
+      else
+        Puppet::Util::Platform.pretend_to_be :linux
+      end
+      Puppet.settings[:autosign] = false
```

Comment:
Preserve the note on Windows validating the value here, and note that the setting is not used during unit-testing anyways.